packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -230,134 +232,249 @@ class GestureDetector extends StatelessWidget {
   final Widget child;
 
   /// A pointer that might cause a tap has contacted the screen at a particular
-  /// location.
+  /// location. Only responds to primary buttons.
   ///
   /// This is called after a short timeout, even if the winning gesture has not
   /// yet been selected. If the tap gesture wins, [onTapUp] will be called,
   /// otherwise [onTapCancel] will be called.
+  ///
+  /// See also:
+  ///
+  ///  * [kPrimaryButton], which is the buttons this callback responds to.
   final GestureTapDownCallback onTapDown;
 
   /// A pointer that will trigger a tap has stopped contacting the screen at a
-  /// particular location.
+  /// particular location. Only responds to primary buttons.
   ///
   /// This triggers immediately before [onTap] in the case of the tap gesture
   /// winning. If the tap gesture did not win, [onTapCancel] is called instead.
+  ///
+  /// See also:
+  ///
+  ///  * [kPrimaryButton], which is the buttons this callback responds to.
   final GestureTapUpCallback onTapUp;
 
-  /// A tap has occurred.
+  /// A tap has occurred. Only responds to primary buttons.
   ///
   /// This triggers when the tap gesture wins. If the tap gesture did not win,
   /// [onTapCancel] is called instead.
   ///
   /// See also:
   ///
+  ///  * [kPrimaryButton], which is the buttons this callback responds to.
   ///  * [onTapUp], which is called at the same time but includes details
-  ///    regarding the pointer position.
+  ///    regarding the pointer position and buttons.
   final GestureTapCallback onTap;
 
   /// The pointer that previously triggered [onTapDown] will not end up causing
-  /// a tap.
+  /// a tap. Responds to any buttons.
   ///
   /// This is called after [onTapDown], and instead of [onTapUp] and [onTap], if
   /// the tap gesture did not win.
   final GestureTapCancelCallback onTapCancel;
 
+  /// A pointer that might cause a tap has contacted the screen at a particular
+  /// location. Only responds to secondary buttons.
+  ///
+  /// This is called after a short timeout, even if the winning gesture has not
+  /// yet been selected.
+  ///
+  /// It does not respond when a stylus touches the screen while its secondary button
+  /// is pressed, since screen touching is counted as a button and fails button
```

Comment:
```suggestion
  /// is pressed, since screen touching is counted as a button and so it fails the button
```

You might also explain what the "button check" prevents.